magma/primitives/when.py

```diff
@@ -149,15 +149,16 @@ def input_to_index(self) -> Dict[Type, int]:
     def output_to_index(self) -> Dict[Type, int]:
         return self._output_to_index.copy()
 
-    def _check_existing_array_child(self, value, value_to_name, value_to_index):
-        """If value is a child of an array that has already been added, we
-        return the child of the existing value, rather than adding a new port,
-        which allows us to maintain bulk assignments in the eventual generated
-        if statement, rather that elaborating into per-child assignments
+    def _check_existing_derived_ref(self, value, value_to_name, value_to_index):
+        """If value is a child of an array or tuple that has already been added,
+        we return the child of the existing value, rather than adding a new
+        port, which allows us to maintain bulk assignments in the eventual
+        generated if statement, rather that elaborating into per-child
+        assignments
```

Comment:
while we're here, nit
```suggestion
        assignments.
```